lib/workos/sso.ex

```diff
@@ -0,0 +1,186 @@
+defmodule WorkOS.SSO do
+  @moduledoc """
+  Manage Single Sign-On (SSO) in WorkOS.
+
+  @see https://docs.workos.com/sso/overview
+  """
+
+  require Logger
+
+  alias WorkOS.Empty
+  alias WorkOS.SSO.Connection
+  alias WorkOS.SSO.Profile
+  alias WorkOS.SSO.ProfileAndToken
+
+  @provider_types ["GoogleOAuth", "MicrosoftOAuth"]
```

Comment:
Going to remove those checks against enums, those should be expected as regular strings, and then the API should return a 400 error if the valid is invalid. This is harder to maintain as we remove/add new enum values from our API and have to update across SDKs.